vendors/swedbank-pay-core-library/src/SwedbankPay/Core/Library/PaymentInfo.php

```diff
@@ -0,0 +1,218 @@
+<?php
+
+namespace SwedbankPay\Core\Library;
+
+use SwedbankPay\Core\Api\Authorization;
+use SwedbankPay\Core\Api\Response;
+use SwedbankPay\Core\Api\Transaction;
+use SwedbankPay\Core\Api\Verification;
+use SwedbankPay\Core\Exception;
+use SwedbankPay\Core\Log\LogLevel;
+
+trait PaymentInfo
+{
+    /**
+     * Do API Request
+     *
+     * @param       $method
+     * @param       $url
+     * @param array $params
+     *
+     * @return Response
+     * @throws \Exception
+     */
+    public function request($method, $url, $params = [])
+    {
+        // Get rid of full url. There's should be an endpoint only.
+        if (filter_var($url, FILTER_VALIDATE_URL, FILTER_FLAG_SCHEME_REQUIRED)) {
+            $parsed = parse_url($url);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The use of function parse_url() is discouraged](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-woocommerce-payments/pullRequest?prid=5340032)